metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -92,13 +94,47 @@ public EbeanEntityService(
   @Override
   @Nullable
   public RecordTemplate getAspect(@Nonnull final Urn urn, @Nonnull final String aspectName, @Nonnull long version) {
+    if (version < 0) {
+      version = _entityDao.getMaxVersion(urn.toString(), aspectName) - version + 1;
```

Comment:
Let's suppose: passed arg version is "-1", `getMaxVersion` returns `2`. In this case line 98 assigns:
`version = 2 - (-1) + 1 = 4`

Is it correct @shirshanka ?

Reply:
Thanks for catching @amorskoy - this was fixed in https://github.com/linkedin/datahub/pull/2723